### ISSUE TITLE
Decorator for .decorator module

### ISSUE THREAD

**Issue by rodrijuarez** [PillowPillow/ng-annotations#7]
Why not adding a @decorator annotation so we can override the angularjs services easily?

**Comment by PillowPillow** [comment]
Hi, i never had a need for an annotation Decorator until now, i'm used to decorate my services inside my config blocks with $provide but it's a good idea to simplify the process.
Do you have a precise idea about the implementation of the Decorator annotation?

**Comment by rodrijuarez** [comment]
That's how I would expect the annotation @decorator to work

```
@decorator('$rootScope') // This is the name of the service that we want to decorate, this should be exported in the $name of the class
@inject('$delegate') // This is the object of the service that we are going to modify
export default class DecoratorClassName {
  constructor ($delegate) {
    //We do modifications to the object here and return the same object with the modifications (more properties, functions, etc)
    return $delegate
  }
}
```

Here is a working example of a decorator using the @service annotation that I've implemented in one of the projects I've been working on:

```
import postal from 'postal'
import {inject, service} from 'ng-annotations'

@service('$rootScope')
@inject('$delegate')
export default class ScopeDecorator {
  constructor ($delegate) {
//Extra code where I do the modifications

    return $delegate
  }
}
```

**Comment by PillowPillow** [comment]
Hi, sorry fot the waiting i needed to think about this new feature.
The component decorator is now available with the v1.1.0, it provides a new way to override the existing services by creating annotated classes.

ex:

``` javascript
import {decorator, inject} from 'ng-annotations';

@decorator('service to decorate')
@inject('$delegate')
export default class DecoratedService {

    constructor($delegate) {
        $delegate.newMethod = () => console.log('hello world');
    }

}
```

Like the factory decorator, an implicit method exist ($decorate) which provides a way to completely override the given service. By Default this method returns a $delegate instance.

Thank's for the issue
Best regards
